examples/api/plot_softplus.py

```diff
@@ -0,0 +1,166 @@
+#     default: 2e-1
+# - `max_iter`: int (maximum iteration for the model)
+#     default: 1000
+
+########################################################
+
+########################################################
+# Import ``GLM`` class from ``pyglmnet``
+
+########################################################
+
+# import GLM model
+from pyglmnet import GLM
+
+# create regularization parameters for model
+reg_lambda = np.logspace(np.log(0.5), np.log(0.01), 10, base=np.exp(1))
+glm_poisson = GLM(distr='softplus', verbose=False, alpha=0.05,
+            max_iter=1000, learning_rate=2e-1,
+            reg_lambda=reg_lambda)
+
+##########################################################
+# Simulate a dataset
+# ------------------
+# The ``GLM`` class has a very useful method called ``simulate()``.
+#
+# Since a canonical link function is already specified by the distribution
+# parameters, or provided by the user, ``simulate()`` requires
+# only the independent variables ``X`` and the coefficients ``beta0``
+# and ``beta``
+
+##########################################################
+
+n_samples, n_features = 10000, 100
+
+# coefficients
+beta0 = np.random.normal(0.0, 1.0, 1)
+beta = sps.rand(n_features, 1, 0.1)
+beta = np.array(beta.todense())
+
+# training data
+Xr = np.random.normal(0.0, 1.0, [n_samples, n_features])
+yr = glm_poisson.simulate(beta0, beta, Xr)
+
+# testing data
+Xt = np.random.normal(0.0, 1.0, [n_samples, n_features])
+yt = glm_poisson.simulate(beta0, beta, Xt)
+
+##########################################################
+# Fit the model
+# ^^^^^^^^^^^^^
+# Fitting the model is accomplished by a single GLM method called `fit()`.
+
+##########################################################
+
+scaler = StandardScaler().fit(Xr)
+glm_poisson.fit(scaler.transform(Xr), yr)
+
+##########################################################
+# Slicing the model object
+# ^^^^^^^^^^^^^^^^^^^^^^^^
+# Although the model is fit to all values of reg_lambda specified by a regularization
+# path, often we are only interested in further analysis for a particular value of
+# ``reg_lambda``. We can easily do this by slicing the object.
+#
+# For instance ``model[0]`` returns an object identical to model but with ``.fit_``
+# as a dictionary corresponding to the estimated coefficients for ``reg_lambda[0]``.
+
+##########################################################
+# Visualize the fit coefficients
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# The estimated coefficients are stored in an instance variable called ``.fit_``
+# which is a list of dictionaries. Each dictionary corresponds to a
+# particular ``reg_lambda``
+
+##########################################################
+
+fit_param = glm_poisson[-1].fit_
+plt.plot(beta[:], 'bo', label ='true')
+plt.plot(fit_param['beta'][:], 'ro', label='estimated')
+plt.xlabel('samples')
+plt.ylabel('outputs')
+plt.legend(bbox_to_anchor=(0., 1.02, 1., .102), loc=1,
+           ncol=2, borderaxespad=0.)
```

Comment:
we don't need publication quality figures in examples. The code should be simple so as to not scare new users :)